stats.go

```diff
@@ -419,6 +450,18 @@ func (s *statStore) NewTimer(name string) Timer {
 	return s.NewTimerWithTags(name, nil)
 }
 
+func (s *statStore) NewTimerWithTag(name, key, value string) Timer {
+	name = serializeTag(name, key, value)
+	if v, ok := s.timers.Load(name); ok {
+		return v.(*timer)
+	}
+	t := &timer{name: name, sink: s.sink}
+	if v, loaded := s.timers.LoadOrStore(name, t); loaded {
+		t = v.(*timer)
+	}
+	return t
+}
+
 func (s *statStore) NewTimerWithTags(name string, tags map[string]string) Timer {
```

Comment:
Can we passthrough to `NewTimerWIthTag` when `len(tags) == 1`? Would that still give us a good chunk of the perf gains? (Same comment everywhere else)